### DISCUSSION TITLE
How can I get the accessToken using Slack Provider?

### DISCUSSION THREAD

**Discussion by yuuu** [nextauthjs/next-auth#2289] [Help] [answered]
## Question 💬

I want to use this npm for OAuth authentication with Slack.
The authentication was successful, but the accessToken was not included in the data that can be retrieved by useSession().

```json
{
  "user": {
    "name": "Firstname Lastname",
    "email": "user@example.com",
    "image": "https://avatars.slack-edge.com/2021-07-01/xxxxxxxxxxxx_xxxxxxxxxxxxxxxxxxxx_512.jpg"
  },
  "expires": "2021-07-30T21:59:33.820Z"
}
```

I would like to send this token to the backend and validate the token with the auth.test API provided by Slack.

How can I get the accessToken?

## How to reproduce ☕️

I have used the following URL to set up the system.

https://next-auth.js.org/providers/slack

**Comment by balazsorban44** [comment, marked as answer]
We only expose a subset of values by default for increased security, but you can configure it. See the docs:

https://next-auth.js.org/configuration/callbacks#jwt-callback

https://next-auth.js.org/configuration/callbacks#session-callback

https://next-auth.js.org/configuration/providers#using-a-custom-provider

https://next-auth.js.org/providers/slack#options

**Reply by yuuu** [reply]
I see. I hadn't found that page.
I'll give it a try. Thank you very much.
